### PR TITLE
[k8s] iotedged: Generate image pull secret for both proxy and edgeagent

### PR DESCRIPTION
**Pull request by dmolokanov**
Initially, we assume that proxy will live in the public container registry that maybe not the case for some customers. So edge should be able to provide all credentials for k8s to pull and run proxy as a sidecar container.
- Refactor `config.yaml` file structure to support proxy settings
- Generate image pull secret for both EdgeAgent and Proxy
- Pass proxy image pull secret to EdgeAgent in order for other modules to have proxy pulled

### CODE REVIEW

File: edgelet/edgelet-kube/src/registry/pull.rs [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+{
+    let image_pull_secrets = vec![config.auth(), runtime.settings().proxy().auth()]
+        .into_iter()
+        .filter_map(|auth| {
+            auth.and_then(|auth| ImagePullSecret::from_auth(&auth))
+                .map(|secret| (secret.name(), secret))
+        })
+        .collect::<HashMap<_, _>>()
+        .drain()
+        .map(|(_, secret)| secret)
```
Comment (darobs) [inline edgelet/edgelet-kube/src/registry/pull.rs]: Possible suggestion: 
```
use itertools::Itertools;

let image_pull_secrets = vec![config.auth(), runtime.settings().proxy().auth()]
        .into_iter()
        .filter_map(|auth| {
            auth.and_then(|auth| ImagePullSecret::from_auth(&auth))
        })
        .unique_by(|secret| secret.name())
        .collect::<Vec<_>>();
```

File: kubernetes/charts/edge-kubernetes/templates/_helpers.tpl [hunk 2]
```diff
@@ -78,13 +78,22 @@ listen:
...
-proxy_config_path: "/etc/iotedge-proxy"
-proxy_config_map_name: "iotedged-proxy-config"
-proxy_trust_bundle_path: "/etc/trust-bundle"
-proxy_trust_bundle_config_map_name: "iotedged-proxy-trust-bundle"
-image_pull_policy: {{ .Values.iotedgedProxy.image.pullPolicy | quote }}
 device_hub_selector: ""
+proxy:
+  image: "{{.Values.iotedgedProxy.image.repository}}:{{.Values.iotedgedProxy.image.tag}}"
+  image_pull_policy: {{ .Values.iotedgedProxy.image.pullPolicy | quote }}
+  {{- if .Values.edgeAgent.registryCredentials }}
```
Comment (darobs) [inline kubernetes/charts/edge-kubernetes/templates/_helpers.tpl]: We are assuming that the proxy registry==agent registry?

File: edgelet/edgelet-kube/src/registry/pull.rs (hunk 1 above)
Comment (dmolokanov) [reply edgelet/edgelet-kube/src/registry/pull.rs]: I like this idea, but do we want to add additional crate?

File: kubernetes/charts/edge-kubernetes/templates/_helpers.tpl (hunk 2 above)
Comment (dmolokanov) [reply kubernetes/charts/edge-kubernetes/templates/_helpers.tpl]: nope. mistake here

File: edgelet/edgelet-kube/src/registry/pull.rs (hunk 1 above)
Comment (darobs) [reply edgelet/edgelet-kube/src/registry/pull.rs]: It's not a required change, I just like the fact I found a more terse way to do the same thing. :)

**Pull request merged**